### PR TITLE
Harmonize plots

### PR DESCRIPTION
**Pull request by namsaraeva**
Most changes include enabling of showing/saving the plots and unifying (some) parameter names across different modules. Some functions got a palette/colorbar arguments. 

Some imports and typos were fixed as well.

### CODE REVIEW

**Comment by namsaraeva** [comment]
todo: add scgen examples 2 and 3

**Comment by namsaraeva** [comment]
scgen examples need to be added, but in a separate pr, i'll open an issue

**Comment by Zethson** [comment]
@namsaraeva could you please write a proper PR description and explain how you handled colors the way you did? On a quick glance I don't see tooo many changes to the color handling in the plots

File: pertpy/plot/_augur.py [hunk 1]
```diff
@@ -95,12 +95,10 @@ def important_features(
...
         ag = Augur("random_forest_classifier")
 
-        return ag.plot_important_features(data=data, key=key, top_n=top_n, ax=ax, return_figure=return_figure)
+        return ag.plot_important_features(data=data, key=key, top_n=top_n, ax=ax)
 
     @staticmethod
-    def lollipop(
-        data: dict[str, Any], key: str = "augurpy_results", ax: Axes = None, return_figure: bool = False
-    ) -> Figure | Axes:
+    def lollipop(data: dict[str, Any], key: str = "augurpy_results", ax: Axes = None) -> Figure | Axes:
```
Comment (Zethson) [inline pertpy/plot/_augur.py:101]: `plot_dp_scatter` for example returns `Figure | Axes | None`. Why does this one not return `| None`?

File: pertpy/tools/_augur.py [hunk 2]
```diff
@@ -1034,16 +1039,24 @@ def plot_dp_scatter(
         legend1 = ax.legend(*scatter.legend_elements(), loc="center left", title="z-scores", bbox_to_anchor=(1, 0.5))
         ax.add_artist(legend1)
 
-        return fig if return_figure else ax
+        if show:
```
Comment (Zethson) [inline pertpy/tools/_augur.py:1042]: These few lines are repeated everywhere and I wonder whether we can extract them to somewhere. Generally not a fan of `_util.py` of any form, but maybe you have a good idea?

File: pertpy/tools/_coda/_base_coda.py [hunk 3]
```diff
@@ -2130,21 +2129,23 @@ def my_layout(node):
...
-                return tree2.render("%%inline", tree_style=tree_style, units=units, w=w, h=h, dpi=dpi)
+                return tree2.render("%%inline", tree_style=tree_style, units=units, w=figsize[0], h=figsize[1], dpi=dpi)
         else:
             if not show_leaf_effects:
                 return tree2, tree_style
 
     def plot_effects_umap(  # pragma: no cover
         self,
-        data: MuData,
+        mudata: MuData,
```
Comment (Zethson) [inline pertpy/tools/_coda/_base_coda.py:2139]:
```suggestion
        mdata: MuData,
```

adata == AnnData
mdata == Mudata

File: pertpy/tools/_coda/_base_coda.py [hunk 4]
```diff
@@ -2130,21 +2129,23 @@ def my_layout(node):
...
         effect_name: str | list | None,
         cluster_key: str,
         modality_key_1: str = "rna",
         modality_key_2: str = "coda",
+        color_map: Colormap | str | None = None,
```
Comment (Zethson) [inline pertpy/tools/_coda/_base_coda.py:2144]: You added these parameters, but where they used? Am I blind?

Comment (Zethson) [reply pertpy/tools/_coda/_base_coda.py:2144]: palette seems to be used, color_map?

**Review by Zethson — COMMENTED** [review]
A lot of really good stuff! This PR is a great improvement.

1. I would be happy if we could reduce the copy pasta. Can you find a nice spot where to put the few lines? Feel free to discuss it with me.
2. The changes to color handling are a bit conservative here. I'm positive that a few barplots etc that don't use colormaps/palettes yet could also use these as parameters. Yes, this would mean adapting plots that are not scanpy plots atm and passing these to e.g. seaborn plots

File: pertpy/tools/_augur.py (hunk 2 above)
Comment (namsaraeva) [reply pertpy/tools/_augur.py:1042]: Hm, I don't have any better ideas than introducting _utils. Maybe we just can keep it as it is for now until we come up with something else. But I agree, it is kinda annoying to have copy pasta at so many places

**Pull request merged**